packages/services/src/__tests__/helpers.test.ts

```diff
@@ -43,3 +43,147 @@ describe('first', () => {
     expect(result.quantity).toBe(2);
   });
 });
+
+describe('concatService', () => {
+  it('should concatenate the values of the arguments passed to it', () => {
+    const data = {
+      productIdentifier: [
+        {
+          scheme: 'https://id.gs1.org/gtin',
+          identifierValue: '05012345678900',
+          binding: {
+            type: 'document',
+            assuranceLevel: '3rdParty',
+            reference: 'https://id.gs1.org/gtin/05012345678900/binding',
+          },
+        },
+      ],
+      batchIdentifier: [
+        {
+          scheme: 'https://Cherriesfarm.example.org/batch',
+          identifierValue: '2024001',
+          binding: {
+            type: 'document',
+            assuranceLevel: '3rdParty',
+            reference: 'https://Cherriesfarm.example.org/batch/2024-001/binding',
+          },
+        },
+      ],
+      itemIdentifier: [
+        {
+          scheme: 'https://Cherriesfarm.example.org/item',
+          identifierValue: 'TRF240001',
+          binding: {
+            type: 'document',
+            assuranceLevel: '3rdParty',
+            reference: 'https://Cherriesfarm.example.org/item/TRF-24-0001/binding',
+          },
+        },
+      ],
+    };
+    const args: any = [
+      { type: 'text', value: '(01)' },
+      { type: 'path', value: '/productIdentifier/0/identifierValue' },
+      { type: 'text', value: '(10)' },
+      { type: 'path', value: '/batchIdentifier/0/identifierValue' },
+      { type: 'text', value: '(21)' },
+      { type: 'path', value: '/itemIdentifier/0/identifierValue' },
+    ];
+    const result = concatService(data, ...args);
+    expect(result).toBe('(01)05012345678900(10)2024001(21)TRF240001');
+  });
+});
+
+describe('constructIdentifierString', () => {
```

Comment:
Hi, can you add the test case for Invalid Path in identifierKeyPath, Invalid Handler Function and handler function is found but throws an error?

Reply:
good catch!